### ISSUE TITLE
ContainerTraits::size and BufferAdapterTraits::increaseBufferSize

### ISSUE THREAD

**Issue by victorstewart** [fraillt/bitsery#80]
thoughts on remaining size to capacity? because it really requests capacity not size. just debugged an infinite loop where i'd implemented these as "size" when really the intention was "capacity"

**Comment by fraillt** [comment]
Yes... you're absolutely right, I cannot explain how it slipped. I just don't know how should I fix this. 
There are some nice improvements that deserve to be published in a new bistery version. This change is actually a breaking one, so I would want to avoid it... maybe adding `[[deprected]]` attribute to these functions, and creating new ones would be the way to go here.

**Comment by victorstewart** [comment]
meant *rename lol. my mind was focused on remaining.

[[deprecated]] sounds like the least breaking way to fix it. 

but say the version was changed to 6.0, it's not unheard of to have API breaking changes in new major versions. i suspect few people use these facilities since this wasn't noticed before, and they'd realize upon upgrading and compiling.

but either way is fine!
